src/frontend/admin/src/tests/organization/organization.test.e2e.ts

```diff
@@ -44,12 +45,51 @@ test.describe("Organization Form", () => {
     await page.getByLabel("Title").click();
     await page.getByLabel("Title").fill("Organization test");
     await page.getByLabel("Title").fill("Organization title");
-    await page.getByLabel("Code").click();
+    await page.getByLabel("Code", { exact: true }).click();
+    await page.getByLabel("Code", { exact: true }).fill("Organization code");
     await page.getByLabel("Country").click();
     await page.getByRole("option", { name: "United Kingdom" }).click();
-    await page.getByLabel("Code").fill("Organization code");
-    await page.getByLabel("Representative").click();
-    await page.getByLabel("Representative").fill("john.doe@yopmail.com");
+    await page.getByLabel("Representative", { exact: true }).click();
+    await page.getByLabel("Representative", { exact: true }).fill("John Doe");
+
+    await expect(
+      page.getByText("Mandatory if signatory representative is set"),
+    ).toBeVisible();
+
+    await page.getByLabel("Profession of the representative").click();
+    await page.getByLabel("Profession of the representative").fill("Director");
+    await page.getByLabel("Signatory representative", { exact: true }).click();
+
+    await page
+      .getByLabel("Signatory representative", { exact: true })
+      .fill("Doe John");
+    await page.getByRole("button", { name: "Submit" }).click();
+    await expectHaveClasses(
+      page.getByText("signatory_representative_profession is a required field"),
+      "Mui-error",
+    );
+
+    await page.getByLabel("Profession of the signatory").click();
```

Comment:
just a global remark: These tests fill these inputs but we never make sure the submit request to the backends sends accurate data matching these inputs?